loadimpact/k6-docs
# Docs file tree restructuring

**Describe changes**
This PR brings file tree restructuring to `data/markdown` section, which gets all currently missing images back. Side: updated contributing format file.

**Reason**
After merging #71 @ppcano reported missing images on a number of pages. After a short investigation it turned out, that #71 has almost nothing to do with that - the problem is in images position in file tree. I don't know why it even worked before, it really shouldn't have been. 

Here is what I mean:

https://staging.k6.io/docs/using-k6/metrics - one of the reported page. Look at one of the missing images path:
![image](https://user-images.githubusercontent.com/32940211/90612808-fd518e00-e221-11ea-8dbc-136d5fad671b.png)
It requires `custom-metrics.png` in `/images`. But `/images` has no such file:
![image](https://user-images.githubusercontent.com/32940211/90613055-4bff2800-e222-11ea-9cda-216a945ab6fe.png)

That is the cause. 

**Changes, verbose**
- reorganized piles of images in clusters into multiple article-related folders
- checked every image for usage cases - apparently there are plenty of images that are not being used anymore. I've marked them `[unused]-%old_name%`. 
- adjusted a bunch of paths like this: `[](images/...)` -> `[](/images/...)` and replaced some of `%`-including path with kebab-case alternatives. Actually, that is interesting: I have no idea why, but some of the images are perfectly working with path like `[image](images/01%image%folder/image.png)`, but in other folders they just won't and require to be adjusted to `[image](/images/01-image-folder/image.png)`. I've reflected this behaviour in contributing file format. 
- finally I've checked every single page in dev and prod env to make sure no broken images are left

**Steps to test**

- check that I didn't make some severe mutations to articles content somewhere
- pull the changes, run `npm run build && npm run serve`
- go to `localhost:9000/docs`
- random check articles, see that images are indeed there
